test/e2e/specs/fakerp/cvepatch.go

```diff
@@ -0,0 +1,112 @@
+package fakerp
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/ghodss/yaml"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp"
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/sanity"
+	logger "github.com/openshift/openshift-azure/test/util/log"
+)
+
+var _ = Describe("Apply CVE hot patch E2E tests [ApplyHotPatch][Fake][LongRunning]", func() {
+	var (
+		ctx       = context.Background()
+		patchRpms = []string{
+			"python-libs-2.7.5-80.el7_6.x86_64",
+			"python-2.7.5-80.el7_6.x86_64",
+			"vim-minimal-7.4.160-6.el7_6.x86_64",
+		}
+		log = logger.GetTestLogger()
+		// Do not use the internal config in other tests! This is necessary here in order to acquire
+		// ssh credentials which will be used to query the sample vm for rpm lists before and
+		// after cve hot patches are applied to a cluster. Usually fakerp tests should be written
+		// to use the admin config wherever possible.
+		internalConfig = func() (*api.OpenShiftManagedCluster, error) {
```

Comment:
haven't found a better way to grab the list of rpms before and after the hot patch as well as grabbing logs. This is why I resorted to this antipattern of using the internal config in an e2e test. Ideas for improvement welcome.

Reply:
Part of the problem may coincide with needing to improve the error handling/RPM version reporting side of this.